cli/cluster/cmd/replicaset.go

```diff
@@ -73,18 +73,24 @@ type PromoteCtx struct {
 }
 
 // pickPatchKey prompts to select a key to patch the config.
+// pathMsg is a path destination in config to patch. If it is not
+// empty path displays for user.
 // If force is true, picks the first passed key.
-func pickPatchKey(keys []string, force bool) (int, error) {
+func pickPatchKey(keys []string, force bool, pathMsg string) (int, error) {
```

Comment:
I don't really like this change, it makes the code too dependent on callers. I would prefer a specific implementation here for a specific need. On the other hand, the code is simple now, so up to you.

Reply:
Do you mean separate function `pickPatchKeyWithPathMsg` that will add the log message with path in config under the specific implementation?

Reply:
Yeah, something like this.

Reply:
I thought duplicating code was not a pretty good solution. Finally add special struct with enumerated options for `pickPatchKey`. Now it doesn't depend on callers because we may not to push `pathMsg` field in argument to it.

Reply:
Oh, it's better to return it to how it was. The `Opts` argument even worse because it hides arguments and it even harder to read.

Reply:
Put it back the way it was.